### ISSUE TITLE
Full incompatibility with Pale Moon since 1.0.2

### ISSUE THREAD

**Issue by Nindaleth** [EFForg/privacybadgerfirefox#595]
Privacy Badger was installable on Pale Moon as of around 1.0.0 (with the "only" bug being #479), but the current update doesn't allow the installation at all.

Tested Pale Moon versions are 25.5-25.7. bitness and OS doesn't matter.

**Comment by cooperq** [comment]
Can you tell me what errors show up in the browser console?

**Comment by Nindaleth** [comment]
Just the pop-up "Privacy Badger could not be installed because it is not compatible with Pale Moon 25.7."

There are no errors in browser console as far as I can tell (used Ctrl+Shift+I -> Console, cleared before installing)

**Comment by LimboSlam** [comment]
Yes I can confirm this. Also, you should read this: https://forum.palemoon.org/viewtopic.php?f=1&t=9361, as Pale Moon v26.0.0 of the Goanna engine will be release soon. Yeah last I check our new milestone was at 74%.

**Comment by cypherpunk** [comment]
This particular issue is with install.rdf. PaleMoon no longer has the same uuid as firefox, so you need:


    <em:targetApplication>
      <Description>
        <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
        <em:minVersion>26.0</em:minVersion>
        <em:maxVersion>30.0</em:maxVersion>
      </Description>
      <Description>
        <em:id>{8de7fcbb-c55c-4fbe-bfc5-fc555c87dbc4}</em:id>
        <em:minVersion>26.0</em:minVersion>
        <em:maxVersion>39.0</em:maxVersion>
      </Description>
    </em:targetApplication>

**Comment by LimboSlam** [comment]
@cypherpunk: I'm not coder (a good one), but will this also work for Pale Moon v25+ as see only <em:minVersion>26.0</em:minVersion>

Or does this not matter as long as Pale Moon GUID is their install.rdf?

**Comment by VeronicaGarcia** [comment]
@cooperq: Thanks for having faith!  :) :+1:  As well as everyone else.  

Just to keep everyone informed with Pale Moon progress with Goanna: https://forum.palemoon.org/viewtopic.php?f=24&t=9077 and: https://github.com/MoonchildProductions/Pale-Moon/milestones is at 84% now and it's separate release notes for Pale Moon v26.0.0 Pre-release/Goanna: https://www.palemoon.org/releasenotes-goanna.shtml.

**Comment by cooperq** [comment]
@VeronicaGarcia Am I correct in assuming that palemoon will probably never implement firefox's upcoming web extensions api?

**Comment by VeronicaGarcia** [comment]
@cooperq: I would think so. @wolfbeast or @LimboSlam are we right?

**Comment by wolfbeast** [comment]
Web Extensions is not on our roadmap at the moment. We already have XUL, bootstrap and SDK options; I don't particularly see a reason why we should support a 4th extension format that isn't native to our framework to begin with, and seems very limited in what it can actually do in comparison.

Some things will depend on if the Firefox implementation of Web Extensions is easily portable or not, and if it is robust or not. Currently looks **very unlikely** that we'll adopt it.

**Comment by cooperq** [comment]
In that case I want to give you a heads up that once web extensions are stable we are planning to use that API for privacy badger in firefox. At that time we will stop supporting the jetpack version (this version) of privacy badger. Someone else will need to take it over if they want it to keep using privacy badger in Pale Moon. The reason for this is that by using WEAPI we would only have to support one code base instead of two, which cuts the amount of work needed to fix bugs and add features in half. 
I'm not sure what the current timeline is for WEAPI.

**Comment by wolfbeast** [comment]
Perfectly understandable. We are in the same boat -- adding support for the WEAPI would add having to support yet another framework (which is certainly a lot more complex and time-consuming than any extension maintenance would be for an otherwise stable SDK, I think). If you stop supporting the SDK version, then one of our community members will have to fork it and maintain it from there on out.

For future reference when this happens, would it be OK for this maintainer to continue to carry the same name of the extension?

I do hope that until such time as the WEAPI has landed in release versions, you will consider keeping an SDK version available for Pale Moon users as well as Firefox users, to get back on-topic for this issue.

**Comment by grumpygeek** [comment]
No need to fork Privacy Badger.  uMatrix and uBlockOrigin work and are superior to this extension.  My first taste of Privacy Badger was when I had a firefox browser version from Comodo.  And even though Comodo is supposedly in the internet security business (I used their old firewall many years ago), they are now using their software to spy on or track their users - absolutely unacceptable for a company of this type.  I don't recall exactly what Privacy Badger did, or whether it was just Comodo's version of it, but I do recall that I did not consider it useable due to security/privacy concerns.  Anyway, I also lost any trust or respect for EFF that I may have ever had if any over this and the HTTPSEverywhere fiasco.

**Comment by cooperq** [comment]
@wolfbeast  as of the next release we are switching to web extensions and this code base is officially unmaintained. I would definitely welcome a community fork to keep support for pale moon if that is something you are interested in doing!

**Comment by wolfbeast** [comment]
@cooperq Thanks, I'll make a fork into our add-ons organization then and hope someone will pick it up!
